### PR TITLE
Added .ply exporter

### PR DESCRIPTION
**Pull request by friesofdoom**
Added default color lookup in g3d class

### CODE REVIEW

File: csharp/Vim.G3d/G3DPlyExporter.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                }
+
+                writer.Write(StringToByteArray("\n"));
+            }
+
+            //Close the binary writer
+            writer.Close();
+        }
+
+        private static byte[] StringToByteArray(string theString)
```
[the remark this answers is not in the document]
Comment (friesofdoom) [reply csharp/Vim.G3d/G3DPlyExporter.cs]: This doesn't take care of the ASCII translation and you end up with some garbage characters.

File: csharp/Vim.G3d/G3D.cs [hunk 2]
```diff
@@ -80,7 +82,7 @@ public G3D(IEnumerable<BinaryAttribute> attributes, Header header = null)
                         break;
                 }
 
-                if (attr.Descriptor.Semantic == SemanticEnum.sem_position && attr.Descriptor.Association == AssociationEnum.assoc_corner)
+                if (attr.Descriptor.Semantic == SemanticEnum.sem_position && (attr.Descriptor.Association == AssociationEnum.assoc_corner || attr.Descriptor.Association == AssociationEnum.assoc_vertex))
```
[the remark this answers is not in the document]
Comment (friesofdoom) [reply csharp/Vim.G3d/G3D.cs]: Two other things i noticed:
1. you only have one pre-defined Position header which uses per vertex position,
2. the NumVertices etc. isnt initialized correctly

File: csharp/Vim.G3d/G3DPlyExporter.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+using System.Collections.Generic;
+using System.Text;
+using System.IO;
+
+namespace Vim.G3d
+{
+    // A very simple .ply file exporter
+    public static class G3DPlyExporter
+    {
+        public static void WritePly(this G3D g, string filePath)
```
[the remark this answers is not in the document]
Comment (friesofdoom) [reply csharp/Vim.G3d/G3DPlyExporter.cs]: Originally I was going to add a binary ply file path, I might still do that.
I have added an extension to the binary writer to write the ascii.

**Comment by cdiggins** [comment]
Great, thanks a lot for doing this!

**Pull request merged**